crates/services/p2p/src/service.rs

```diff
@@ -555,8 +595,38 @@ where
                 response,
                 request_id,
             },
+            self.max_headers_per_request,
         )
     }
+
+    fn handle_all_transactions_ids_request(
+        &mut self,
+        request_id: InboundRequestId,
+    ) -> anyhow::Result<()> {
+        let response_channel = self.request_sender.clone();
+
+        let tx_ids = self.tx_pool.get_all_tx_ids();
+        response_channel.try_send(TaskRequest::TxPoolAllTransactionsIds {
+            response: tx_ids,
+            request_id,
+        })?;
+        Ok(())
+    }
+
+    fn handle_full_transactions_request(
+        &mut self,
+        tx_ids: Vec<TxId>,
+        request_id: InboundRequestId,
+    ) -> anyhow::Result<()> {
+        let response_channel = self.request_sender.clone();
+
+        let txs = self.tx_pool.get_full_txs(tx_ids);
+        response_channel.try_send(TaskRequest::TxPoolFullTransactions {
+            response: txs,
+            request_id,
+        })?;
+        Ok(())
```

Comment:
Potentially TxPool can be very busy with current work and answering can take a lot of time. It is better to use `handle_request` as well

Reply:
I don't really know what you mean by your first sentence. Also in your second setence you want me to use the thread pool to make the ask to the TxPool ? (I can't directly reuse `handle_request` as it's specifically done for range asks (I can rework the function)

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2131